src/components/GradeLevelFilter.test.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { Suspense } from 'react';
+import { render, screen, waitFor } from '@testing-library/react';
+import GradeLevelFilter from './GradeLevelFilter';
+import { I18nextProvider } from 'react-i18next';
+import i18n from '../i18n';
+import { act } from 'react-dom/test-utils';
+
+function renderComponent(grades = new Set()) {
+  return render(
+    <Suspense fallback="loading">
+      <I18nextProvider i18n={i18n}>
+        <GradeLevelFilter grades={grades} onChange={() => {}} />
+      </I18nextProvider>
+    </Suspense>
+  );
+}
+
+test('renders grade groups', async () => {
+  renderComponent();
+
+  await waitFor(() =>
+    expect(screen.getByText('Middle School')).toBeInTheDocument()
+  );
+  expect(screen.getByText('High School')).toBeInTheDocument();
+  expect(screen.getByText('College/University')).toBeInTheDocument();
+  expect(screen.getByText('Postgraduate')).toBeInTheDocument();
+});
+
+test('renders translated grade groups - Spanish', async () => {
+  renderComponent();
+  await act(() => i18n.changeLanguage('es').then());
+
+  expect(screen.getByText('Secundaria')).toBeInTheDocument();
+  expect(screen.getByText('Preparatoria')).toBeInTheDocument();
+  expect(screen.getByText('Colegio/Universidad')).toBeInTheDocument();
+  expect(screen.getByText('Postgrado')).toBeInTheDocument();
+});
+
+test('renders checkmarks for applicable grades', async () => {
+  renderComponent(new Set([9, 13, 18]));
+
+  await waitFor(() =>
+    expect(screen.getByRole('checkbox', { name: '9th Grade' })).toBeChecked()
+  );
+  expect(screen.getByRole('checkbox', { name: 'Freshman' })).toBeChecked();
+  expect(screen.getByRole('checkbox', { name: '1st Year' })).toBeChecked();
```

Comment:
another way to do this might be `getAllByRole`, check the length is 0 and then check mapping to the name matches these values.

Reply:
tried the following: 
```
 const checkedGrades = screen.getAllByRole('checkbox', { checked: true });
  expect(checkedGrades).toHaveLength(3);
  expect(checkedGrades[1]).toHaveTextContent('9th Grade');
```

the 2nd statement worked but the last one didn't due to the textContent being empty. Not sure how to get the name for each checkbox.

Reply:
Does `toHaveAttribute`, `innerText`, or `innerHTML` work? Probably the easiest thing is to mistype the role so that you see the HTML itself and can identify if there's something there you can use. Maybe you can just check the `value` attribute or something

Reply:
hmm I've tried different ways to try and access the name but everytime it comes back as null, '', or it just shows a blank.